### ISSUE TITLE
Add a "system" right to replace checks for SEC_inGroup('root')

### ISSUE THREAD

**Issue by leegarner** [glFusion/glfusion#383]
To help delegate site administration to other users/groups without confusing group membership, add a "system" right such as "system.root" to be used instead of checking for Root membership.

Example, I have a site with a group "Club Officers". As the administrator, I may not be an officer but as a member of Root I am included in the group listing of officers and other groups anyway. If my account could be removed from Root but granted the system.root privilege, then all root access would be available without being a member of every other site group.

There are other ways, such as looking for directly-assigned members for the group list, but it seems simpler to avoid the group inclusion of Root completely.

For instance: 
```
--- a/private/system/classes/config.class.php
+++ b/private/system/classes/config.class.php
@@ -792,7 +792,7 @@ class config
         if(!array_key_exists($grp, $LANG_configsubgroups)) {
             $LANG_configsubgroups[$grp] = array();
         }
-        if (!SEC_inGroup('Root')) {
+        if (!SEC_hasRights('system.root')) {
             return config::_UI_perm_denied();
         }
```

**Comment by mark0263** [comment]
What if - we modified SEC_inGroup() whenever root is passed - we also check for system.root feature. This way, you get all root access without being a member of root.  Otherwise, would probably have to hit all the code where SEC_inGroup('Root') is called.

**Comment by leegarner** [comment]
Sure, that would be great.

**Comment by leegarner** [comment]
The main thing I'd like to avoid is the automatic membership that Root memberrs have in all other groups. So the "right" wouldn't technically be necessary, but conceptually I'd think that Rights are better suited for feature access and Groups for content access.

**Comment by mark0263** [comment]
the other option is I can bite the bullet and build out permissions / rights for the features that check for root.  Looks like:
 - Admin Actions
 - Clear Cache
 - Configuration
 - Database Admin
 - Environment Check
 - Logview
 - SFS User Check
 - Upgrade Check

So I could create a feature / right for each - actions.admin, cache.admin, config.admin, database.admin, env.admin, logview.admin, sfs.admin, upgrade.admin.

Then we have the ability to be a bit more granular on granting permissions to certain areas.

I did code the system.root and it works fine - with just that right, you get the items listed above.

**Comment by walterrowe** [comment]
I wonder how many sites would really grant those rights to that level of granularity?

**Comment by leegarner** [comment]
For now I think that's fine. I also have a feature class that I've worked
on to add more management for features/rights.  I think for the things that
you mention a single "system.root" feature (or in group Root) is fine. I
don't see very many site owners wanting to separate admin duties that
granularly.

**Comment by leegarner** [comment]
LOL, Walter had the same thought while I was typing...

**Comment by mark0263** [comment]
system.root capability implemented with 3b81d5f4e683e1ceb76aed9964867e71234cced9
